lib/ueberauth/strategy/google.ex

```diff
@@ -130,15 +125,21 @@ defmodule Ueberauth.Strategy.Google do
     end
   end
 
-  defp with_param(opts, key, conn) do
-    if value = conn.params[to_string(key)], do: Keyword.put(opts, key, value), else: opts
-  end
-
-  defp with_optional(opts, key, conn) do
-    if option(conn, key), do: Keyword.put(opts, key, option(conn, key)), else: opts
+  @doc false
+  defp redirect_options(conn) do
```

Comment:
I don't think I like this refactor, for two reasons:

+ What the code was doing was clear previously
+ This requires the use of `to_string` on all the keys, many of which don't exist as params

I'm open to supporting additional parameters but I think there is likely a more efficient way to do so.  What do you think?